src/context_free.ml

```diff
@@ -503,18 +524,24 @@ class map_top_down ?(expect_mismatch_handler = Expect_mismatch_handler.nop)
               | None ->
                   self#pexp_apply_without_traversing_function base_ctxt e func
                     args
-              | Some e -> self#expression base_ctxt e))
+              | Some e -> (
+                  try self#expression base_ctxt e
```

Comment:
Why do you catch exceptions here? I would not expect this to raise, these are just calls to "continue the traverse", not an expansion. (There are similar `try with` on such expression later!)